sphinx/writers/latex.py

```diff
@@ -809,7 +809,7 @@ def generate_numfig_format(self, builder):
             ret.append(self.babel_renewcommand('\\tablename', definition))
             if table[1]:
                 ret.append('\\makeatletter\n')
-                ret.append('\\def\\fnum@table{\\tablename\\thetable%s}\n' %
+                ret.append('\\def\\fnum@table{\\tablename\\thetable\\relax{}%s}\n' %
```

Comment:
same remark, ``\thetable %s`` fixes it too and is simpler. Now if `%s` was a space this would not give a space in output. But currently `%s` is stripped (which I find is disputable) so it does not matter. Probably best is ``\thetable{}`` (no need for ``\relax``, only ``{}``) because if `%s` is a space it will be obeyed. I think LaTeX should behave like HTML and in my brief testing HTML does not strip spaces.